middle_end/flambda2/types/grammar/type_grammar.ml

```diff
@@ -2690,20 +2737,21 @@ module Row_like_for_blocks = struct
     | Bottom -> (
       match Tag.Map.get_singleton known_tags with
       | None -> None
-      | Some (tag, { maps_to; index; env_extension = _ }) -> (
+      | Some (_tag, Unknown) -> None
```

Comment:
```suggestion
      | Some (_tag, Unknown) -> None
      (* CR pchambart: We lose the tag information when we don't know the shape. Example where this could matter: in Provers.prove_physical_equality this could miss some physical inequality *)
```